utils/compile-time-utils/src/lib.rs

```diff
@@ -20,39 +21,51 @@ pub fn build_date() -> &'static str {
 }
 
 /// Returns Github Actions build number if available or None.
-pub fn build_number_str() -> Option<&'static str> {
+pub fn build_number() -> Option<&'static str> {
     option_env!("GITHUB_RUN_NUMBER")
 }
 
-pub fn build_number() -> Option<u64> {
-    build_number_str().map(|i| {
-        // should not panic
-        i.parse().unwrap()
-    })
-}
-
 /// convert tag to a semantic version
-pub fn semver_str() -> &'static str {
+pub fn semver_str() -> String {
     let mut version = git_tag();
     for prefix in ["pre-rel-", "v"].iter() {
         if version.starts_with(prefix) {
             version = &version[prefix.len()..];
         }
     }
-    version
+    if let Some(bn) = build_number() {
+        [version, bn].join("+")
```

Comment:
I don't think it's a good idea to smuggle in a change in version format. Build number is already included in `version_describe`.